mysql-test/main/distinct.test

```diff
@@ -928,4 +928,47 @@ drop table t1;
 
 --echo #
 --echo # end of 10.5 tests
+
+--echo #
+--echo # MDEV-30660 COUNT DISTINCT seems unnecessarily slow when run on a PK
+--echo #
+
+CREATE TABLE t1 (a INT NOT NULL PRIMARY KEY, b INT NOT NULL);
+INSERT INTO t1 VALUES (1,1), (2,1), (3,1);
+
+SELECT COUNT(DISTINCT a) FROM t1;
+SELECT COUNT(DISTINCT (SELECT a)) FROM t1;
+SELECT AVG(DISTINCT a) FROM t1;
+SELECT SUM(DISTINCT a) FROM t1;
+SELECT SUM(DISTINCT a) FROM t1 WHERE a > 1;
+SELECT SUM(DISTINCT a + b) FROM t1;
```

Comment:
this case doesn't use the optimization as `func->arguments()[i]->type()` is `Item::FUNC_ITEM`. Same with last `AVG(DISTINCT b + a)`. For a PK(a) this is correct, but should these be listed separately to show the not-selected case.

Reply:
OK, moved them to a separate part of the test file and marked them with `--echo #`